www/js/app.js

```diff
@@ -886,6 +904,60 @@ function initServiceWorkerMessaging () {
     }
 }
 
+/**
+ * Function that handles a message of the messageChannel.
+ * It tries to read the content in the backend, and sends it back to the ServiceWorker
+ *
+ * @param {Event} event The event object of the message channel
+ */
+async function handleMessageChannelByWasm (event) {
+    // We received a message from the ServiceWorker
+    // The ServiceWorker asks for some content
+    const title = event.data.title;
+    const messagePort = event.ports[0];
+    try {
+        const ret = await selectedArchive.getEntryDirByWasm(title);
+        if (ret === null) {
+            console.error('Title ' + title + ' not found in archive.');
+            messagePort.postMessage({ action: 'giveContent', title: title, content: '' });
+            return;
+        }
+
+        if (ret.mimetype === 'unknown') {
```

Comment:
I couldn't get the `redirect` variable as true for `W/mainPage`

Reply:
Do you mean the dirEntry property `isredirect`? Does libzim return that property or something similar? It might be a function rather than a property (`isredirect()`). Worth checking by pausing when the dirEntry is returned and inspecting its properties. Some are different from the ones we're used to. E.g. mimetype is a string rather than a function. Come to think of it, I'm not sure we can return functions over postMessage, but I may be wrong.

Reply:
After calling `await this.callLibzimWorker({ action: 'getEntryByPath', path: "W/mainPage" })` We get the result back with mime-type as `unknown` and redirect as `false`. Basically, libzim is unable to tell if the requested title is a redirect or not so if libzim does return false. I am checking if it's a redirect or not by `selectedArchive.resolveRedirect`

Reply:
`await this.callLibzimWorker({ action: 'getEntryByPath', path: "W/mainPage" })` returns this 
`{
    "content": {},
    "mimetype": "unknown",
    "isRedirect": false
}`

Reply:
OK, that's strange. It should show as a redirect when examining the dirEntry, because W/mainPage is supposed to be a redirect to an actual HTML (usually) entry. I'll see if I can troubleshoot this.